Sources/PayPalMessages/Enums/Environment.swift

```diff
@@ -78,26 +71,53 @@ public enum Environment: Equatable {
         case log = "/v1/credit/upstream-messaging-events"
     }
 
-    func url(_ path: PayPalMessagePath, _ queryParams: [String: String?]? = nil) -> URL? {
-        var parts = URLComponents()
-        var queryItems: [URLQueryItem]?
+    private var devTouchpoint: Bool {
+        switch self {
+        case .stage(_, let devTouchpoint, _):
+            return devTouchpoint
+        default:
+            return false
+        }
+    }
 
-        if let queryParams, !queryParams.isEmpty {
-            queryItems = queryParams.map { URLQueryItem(name: $0.key, value: $0.value) }
+    private var stageTag: String? {
+        switch self {
+        case .stage(_, _, let stageTag):
+            return stageTag
+        default:
+            return nil
         }
+    }
+
+    func url(_ path: PayPalMessagePath, _ queryParams: [String: String?]? = nil) -> URL? {
+        var parts = URLComponents()
+        var queryItems = queryParams?.compactMap { key, value in
+            value != nil ? URLQueryItem(name: key, value: value) : nil
+        } ?? []
 
         let basePath: URL
         if path == .log {
             basePath = loggerBaseURL
         } else {
             basePath = baseURL
+
+            // Append dev_touchpoint and stage_tag query parameters only for .stage case
+            if case .stage(_, let devTouchpoint, let stageTag) = self {
+                queryItems.append(URLQueryItem(name: "dev_touchpoint", value: "\(devTouchpoint)"))
```

Comment:
Can we only append this when the value is `true` so that reduces noise in the query params when it's not being used?